CDL-Dryad/stash
# Payment system -- Stripe invoicing

Add communication with the Stripe API to generate invoices for data deposits. Addresses issue  https://github.com/CDL-Dryad/dryad-product-roadmap/issues/190.

An invoice is generated whenever an dataset's curation status is set to "Published" or "Embargoed", and the dataset's Identifier has not yet been invoiced.

This PR generates an invoice for **all** datasets. It does not include the logic for when a dataset is being paid by a journal or institution. That logic will be part of https://github.com/CDL-Dryad/dryad-product-roadmap/issues/191.

To deploy:
- This PR must be deployed in conjunction with https://github.com/CDL-Dryad/dryad/pull/48 and https://github.com/cdlib/dryad-config/pull/19. 
- Also note that there is a database migration required.

To test: 
1. Change any dataset to curation status "Published" or "Embargoed".
2. Look in the database and verify that the submitter's entry in stash_engine_users now has a Stripe customer_id, and the dataset's entry in stash_engine_identifiers now has a Stripe invoice_id.
3. If you have a Stripe account, you can view the invoice there. (The invoice won't actually be sent to the user, since Stripe is in test mode.)
